File: src/unity/Runtime/AdMob/IAdMob.cs
using System.Threading.Tasks;

namespace EE {
    public interface IAdMob : IPlugin {
        Task<bool> Initialize();
        string GetEmulatorTestDeviceHash();
        void AddTestDevice(string hash);
        IBannerAd CreateBannerAd(string adId, AdMobBannerAdSize adSize);
        IFullScreenAd CreateAppOpenAd(string adId);
        IFullScreenAd CreateInterstitialAd(string adId);
        IFullScreenAd CreateRewardedInterstitialAd(string adId);
        IFullScreenAd CreateRewardedAd(string adId);
    }
}

File: src/ts/src/ad_mob/IAdMob.ts
import {
    IBannerAd,
    IFullScreenAd,
} from "../ads";
import { IPlugin, ILibraryAnalytics } from "../core";
import { AdMobBannerAdSize } from "./AdMobBannerAdSIze";

export interface IAdMob extends IPlugin {
    initialize(analytics: ILibraryAnalytics): Promise<boolean>;
    getEmulatorTestDeviceHash(): string;
    addTestDevice(hash: string): void;
    openTestSuite(): void;
    createBannerAd(adId: string, adSize: AdMobBannerAdSize): IBannerAd;
    createAppOpenAd(adId: string): IFullScreenAd;
    createInterstitialAd(adId: string): IFullScreenAd;
    createRewardedInterstitialAd(adId: string): IFullScreenAd;
    createRewardedAd(adId: string): IFullScreenAd;
}